lib/widgets/subscription_list.dart

```diff
@@ -206,11 +231,18 @@ class SubscriptionItem extends StatelessWidget {
       color: designVariables.background,
       child: InkWell(
         onTap: () {
-          Navigator.push(context,
-            MessageListPage.buildRoute(context: context,
-              narrow: ChannelNarrow(subscription.streamId)));
+          final narrow = ChannelNarrow(subscription.streamId);
+          if (onSelect != null) {
+            onSelect!(narrow);
+          } else {
+            Navigator.push(context,
+              MessageListPage.buildRoute(context: context,
+                narrow: narrow));
+          }
         },
-        onLongPress: () => showChannelActionSheet(context, channelId: subscription.streamId),
+        onLongPress: !disableChannelActionSheet
+          ? () => showChannelActionSheet(context, channelId: subscription.streamId)
+          : null,
```

Comment:
Interesting. What goes wrong without this change, if the user does try to open a channel action sheet?